### DISCUSSION TITLE
Problem with pro server daemon

### DISCUSSION THREAD

**Discussion by Marc2p** [BearWare/TeamTalk5#1422] [Q&A] [answered]
Hello,
I recently downloaded the pro version of the team talk 5 server, for testing.
After running the setup wizard, I can start the server by running the binary, like this: ./tt5prosrv -nd -verbose or ./tt5prosrv -d
No problem with that, the problem is configuring the file init.d/tt5proserver and systemd/tt5proserver.service.

I follow the instructions in the README files in both folders, moving the files to /etc/init.d and /etc/systemd/system and running the systemctl enable tt5proserver command.
When running systemctl start tt5proserver, nothing is displayed in the console, however I can't connect to the server.
Running systemctl status tt5proserver shows the following log:

> Aug 03 18:24:22 ip-172-31-41-144 tt5prosrv[4846]: Type username: ("") Type password: ("")
Aug 03 18:24:22 ip-172-31-41-144 tt5prosrv[4846]: Login failed. Please try again.
Aug 03 18:24:22 ip-172-31-41-144 tt5prosrv[4846]: To use BearWare.dk WebLogin please provide your credentials.
Aug 03 18:24:22 ip-172-31-41-144 tt5prosrv[4846]: Type username: ("") Type password: ("")

In loop until I stop it with systemctl stop tt5proserver

The content of init.d/tt5proserver is:

> `#!/bin/sh
### BEGIN INIT INFO
# Provides:          tt5proserver
# Required-Start:    $local_fs $remote_fs
# Required-Stop:     $local_fs $remote_fs
# Default-Start:     2 3 4 5
# Default-Stop:      
# Short-Description: Start/stop TeamTalk 5 Professional server
# Description:       Setup variables BIN, WORKDIR, PIDFILE and USER
#                    before running this script
### END INIT INFO

# Location of TeamTalk 5 Professional server binary
BIN=/home/ubuntu/ttserver/server/tt5prosrv
# Working directory where tt5prosrv.xml and tt5prosrv.log will be written to
WORKDIR=/home/ubuntu/ttserver/server
# File containing the PID of the TeamTalk server daemon
PIDFILE=/home/ubuntu/ttserver/server/tt5prosrv.pid
# The TeamTalk Professional server should be launched by this user account
USER=ubuntu

. /lib/lsb/init-functions

# Carry out specific functions when asked to by the system
case "$1" in
    start)
	log_daemon_msg "Starting TeamTalk Professional Server " "tt5srv"
	start-stop-daemon --start --quiet --oknodo --background \
	--chdir "$WORKDIR" --pidfile "$PIDFILE" \
	--exec "$BIN" -- -wd "$WORKDIR" -pid-file "$PIDFILE" -d
	log_end_msg $?
	;;
    stop)
        log_daemon_msg "Stopping TeamTalk 5 Professional Server"
	test -f $PIDFILE && kill `cat $PIDFILE` && rm $PIDFILE
	log_end_msg $?
	;;
    restart)
	$0 stop || exit 1
	$0 start || exit 1
	;;
    reload|force-reload)
        log_daemon_msg "Reloading TeamTalk 5 Professional Server"
	test -f $PIDFILE && kill -HUP `cat $PIDFILE`
	log_end_msg $?
	;;
    *)
	echo "Usage: /etc/init.d/tt5proserver {start|stop|restart}"
	exit 1
	;;
esac

exit 0`

The content of tt5proserver.service is:

> `# TeamTalk Professional server systemd unit
#
# To start this service you must have configured a user and group
# both of the name 'teamtalk'. In addition a server configuration 
# file must be located in /etc/teamtalk/tt5prosrv.xml. The user 
# 'teamtalk' must have read/write access to this file. Likewise
# for the server log file which will be located in 
# /var/log/teamtalk/tt5prosrv.log
#

[Unit]
Description=TeamTalk 5 Professional server daemon
After=network.target

[Service]
ExecStart=/home/ubuntu/ttserver/server/tt5prosrv -nd -c /home/ubuntu/ttserver/server/tt5prosrv.xml -l /home/ubuntu/ttserver/server/log/tt5prosrv.log
ExecReload=kill -HUP $MAINPID
Type=simple
User=ubuntu
Group=ubuntu
StandardOutput=syslog
StandardError=syslog

[Install]
WantedBy=multi-user.target`

How can I fix this?
What am I doing wrong?

Thanks

**Comment by lbk2907** [comment, marked as answer]
Hey there.

That happend is because you have BearWare account setup in your server.

Plese run the server in forground by using tt5prosrv -nd command.

When it ask for your BearWare account, key in your user and press anter.
Now, type your password and press enter.
When it ask you to keep the token in the XML file, choose yes and press
Control+C to exit.

Now, you can start the server using systemctl.

**Comment by Marc2p** [comment]
So, do I have to create my own account in bearware.dk or do you mean the
user account called bearware and without a password on the server?

**Comment by lbk2907** [comment]
Create your account in BearWare website and log in when prompted.

Like what I said, it will ask for your BearWare account when you have
create the user called bearware without the password in your server.
Doesn't matter you created from TeamTalk Client or the setup wizard itself
by choosing number 3 in the account management section when you run the
tt5prosrv -wizard command.

**Comment by Marc2p** [comment]
Great, solved. Thank you!

**Comment by lbk2907** [comment]
You're welcome.
